src/components/TextStyleOptions/TextStyleOptions.tsx

```diff
@@ -1,62 +1,55 @@
 /* eslint-disable no-console */
-import { ActionGroup, Item } from "@adobe/react-spectrum";
-import { Text } from "@adobe/react-spectrum";
-import type { Selection } from "@react-types/shared";
+import { Flex, ToggleButton } from "@adobe/react-spectrum";
 import TagBold from "@spectrum-icons/workflow/TagBold";
 import TagItalic from "@spectrum-icons/workflow/TagItalic";
-import React, { FC } from "react";
+import { ColorPicker } from "components/ColorPicker";
+import React, { FC, useState } from "react";
 
 export interface TextStyleProps {
     isBold?: boolean;
     isItalics?: boolean;
-    onChange?: (selectedKeys: (string | number)[]) => void;
+    onChange?: (options: { isBold?: boolean; isItalics?: boolean }) => void;
 }
 
 const TextStyleOptions: FC<TextStyleProps> = ({ isBold, isItalics, onChange }) => {
-    const [selected, setSelected] = React.useState<Selection>(() => {
-        const initialValue: string[] = [];
-
-        if (isBold) {
-            initialValue.push("bold");
-        }
-
-        if (isItalics) {
-            initialValue.push("italics");
-        }
-        console.log(initialValue);
-
-        return new Set(initialValue);
-    });
-
-    const handleSelectionChange = (selectedKeys: Selection) => {
-        setSelected(selectedKeys);
+    const [selectedBold, setSelectedBold] = useState<boolean>(Boolean(isBold));
+    const [selectedItalics, setSelectedItalics] = useState<boolean>(Boolean(isItalics));
+
+    const handleButtonChange = (valueBold: boolean, valueItalics: boolean) => {
+        setSelectedBold(valueBold);
+        setSelectedItalics(valueItalics);
+        isBold = valueBold;
+        isItalics = valueItalics;
         if (onChange) {
-            const values: (string | number)[] = Array.from(selectedKeys);
-            onChange(values);
+            onChange({
+                isBold: valueBold,
+                isItalics: valueItalics,
+            });
         }
     };
 
     return (
-        <div>
-            <ActionGroup
+        <Flex direction="row" gap="size-50">
+            <ToggleButton
+                isQuiet
                 isEmphasized
-                density="compact"
+                aria-label="BoldButton"
+                isSelected={selectedBold}
+                onChange={selectedBold => handleButtonChange(selectedBold, selectedItalics)}
```

Comment:
Super use of anonymous function! Nada más que este variable name de `selectedBold` está confusing porque el valor del state hook arriba también es `selectedBold`. 

El value que el onChange nos está pasando es diferente y `handleButtonChange` internamente se encarga de usar el `setSelectedBold` para actualizar el state.

```suggestion
                onChange={newBoldValue => handleButtonChange(newBoldValue, selectedItalics)}
```